projects/storefrontapp-e2e-cypress/cypress/e2e/regression/product_details/show-real-time-stock-in-PDP.e2e-flaky.cy.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { visitProductPage } from '../../../helpers/coupons/cart-coupon';
+import * as sampleData from '../../../sample-data/inventory-display';
+import { FeaturesConfig } from '@spartacus/core';
+
+export const stockSelector = 'cx-add-to-cart .info';
+
+export const GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS = 'getProductDetails';
+
+export function interceptProductAvailability(productCode: string) {
+  cy.intercept(
+    'GET',
+    `${Cypress.env('OCC_PREFIX')}/apparel-uk-spa/productAvailabilities?filters=${productCode}:ST`
+  ).as(GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS);
+
+  return GET_PRODUCT_AVAILABILITY_ENDPOINT_ALIAS;
+}
+
+export function configureInventoryDisplay(enable: boolean) {
+  cy.cxConfig({
+    cmsComponents: {
+      ProductAddToCartComponent: {
+        data: {
+          inventoryDisplay: enable,
+        },
+      },
+    },
+  });
+}
+
+export function assertInventoryDisplay(
+  productCode: string,
+  alias: string,
+  functionality: string
+) {
+  cy.get(`${alias}`).then((xhr) => {
+    let isInventoryDisplayActive;
+    cy.getCookie('cxConfigE2E')
+      .should('exist')
+      .then((data) => (isInventoryDisplayActive = data.value.includes('true')));
```

Comment:
[MINOR]
Instead of deriving `isInventoryDisplayActive` from a cookie, what about obtaining it from a new 4th function argument?

Why asking:

If I understand correctly, this code is _meant to_ check whether the CMS component `ProductAddToCartComponent` is configured to display the inventory (i.e. whether its has `inventoryDisplay` in its payload (or Spartacus CMS static configuration).

However what it does is it checks whether the `cxConfig` object (dynamic config chunk injected from e2e tests to the application) contains ANY occurence of a substring `true`.

IMHO it's error prone. For example, this will yield a false positive result, when cxConfig will contain any other Spartacus config property with a value `true` (e.g. enabling some feature toggle).

I can understand this code is copied from other place https://github.com/SAP/spartacus/blob/f398c42658121964b186d820f57b6177e38d71cd/projects/storefrontapp-e2e-cypress/cypress/helpers/inventory-display.ts#L43-L46

But especially because we're dealing here with feature toggle `showRealTimeStockInPDP: true` in cxConfig too, I think we should make the logic more robust.

Reply:
Done